libraries/AP_BattMonitor/AP_BattMonitor_SMBus.cpp

```diff
@@ -18,6 +38,22 @@ bool AP_BattMonitor_SMBus::read_temp(void)
     return false;
 }
 
+// reads the serial number if it's not already known
+// returns true if the read was successful or the number was already known
+bool AP_BattMonitor_SMBus::read_serial_number(void) {
+    uint16_t data;
+
+    // don't recheck the serial number if we already have it
+    if (_serial_number != 0) {
```

Comment:
I think it is. I actually had the thought while working on it, then forgot to correct it. Will address that real quick.